typescript/agentkit/src/wallet-providers/svmKeypairWalletProvider.ts

```diff
@@ -0,0 +1,110 @@
+import { SvmWalletProvider } from "./svmWalletProvider";
+import { Network } from "../network";
+import { Connection, Keypair, PublicKey, VersionedTransaction, LAMPORTS_PER_SOL, SystemProgram, MessageV0, ComputeBudgetProgram } from "@solana/web3.js";
+import bs58 from "bs58";
+import { SOLANA_MAINNET_CHAIN_ID, SOLANA_MAINNET_NETWORK_ID, SOLANA_PROTOCOL_FAMILY, SOLANA_MAINNET_GENESIS_BLOCK_HASH, SOLANA_TESTNET_CHAIN_ID, SOLANA_TESTNET_NETWORK_ID, SOLANA_DEVNET_GENESIS_BLOCK_HASH, SOLANA_DEVNET_CHAIN_ID, SOLANA_DEVNET_NETWORK_ID, SOLANA_TESTNET_GENESIS_BLOCK_HASH } from "../network/svm";
+
+export class SvmKeypairWalletProvider extends SvmWalletProvider {
+    #keypair: Keypair;
+    #connection: Connection;
+    #genesisHash: string;
+
+    /**
+     * Creates a new SvmKeypairWalletProvider
+     * @param keypair - Either a Uint8Array or a base58 encoded string representing a 32-byte secret key
+     * @param rpcUrl - URL of the Solana RPC endpoint
+     */
+    constructor({
+        keypair,
+        rpcUrl,
+        genesisHash,
+    }: {
+        keypair: Uint8Array | string,
+        rpcUrl: string,
+        genesisHash: string,
+    }) {
+        super();
+
+        this.#keypair = typeof keypair === "string" ? Keypair.fromSecretKey(bs58.decode(keypair)) : Keypair.fromSecretKey(keypair);
+        this.#connection = new Connection(rpcUrl);
+        this.#genesisHash = genesisHash;
+    }
+
+    getAddress(): string {
+        return this.#keypair.publicKey.toBase58();
+    }
+
+    getNetwork(): Network {
+        if (this.#genesisHash === SOLANA_MAINNET_GENESIS_BLOCK_HASH) {
+            return {
+                protocolFamily: SOLANA_PROTOCOL_FAMILY,
+                chainId: String(SOLANA_MAINNET_CHAIN_ID),
+                networkId: SOLANA_MAINNET_NETWORK_ID,
+            };
```

Comment:
nit: can we make `Network` typed objects for each of these and a map from `genesisHash` to network object